src/object_cmd.cpp

```diff
@@ -312,10 +312,13 @@ CommandCost CmdBuildObject(TileIndex tile, DoCommandFlag flags, uint32 p1, uint3
 	}
 
 	int hq_score = 0;
+	int build_object_size = 0;
+	Company *c = nullptr;
 	switch (type) {
 		case OBJECT_TRANSMITTER:
 		case OBJECT_LIGHTHOUSE:
 			if (!IsTileFlat(tile)) return_cmd_error(STR_ERROR_FLAT_LAND_REQUIRED);
+			build_object_size = 1;
```

Comment:
Lighthouses and transmitters also have an objectspec, so you can do `build_object_size = size_x * size_y;` outside of the `switch`.